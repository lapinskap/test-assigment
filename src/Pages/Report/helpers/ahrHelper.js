export const getAhrUrl = (url, isAhr) => {
  let result = isAhr ? '/ahr' : '';
  result += url;

  return result;
};

export const asd = () => {};
