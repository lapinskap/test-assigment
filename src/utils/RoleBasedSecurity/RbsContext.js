import React from 'react';
import UserInfo from './UserInfo';

const RbsContext = React.createContext({
  isLoggedIn: false,
  login: () => {},
  logout: () => {},
  userInfo: new UserInfo({}),
});
export default RbsContext;
