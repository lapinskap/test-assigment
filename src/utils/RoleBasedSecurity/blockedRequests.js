export const BLOCKED_REQUESTS_STORAGE_KEY = 'blocked_requests';

const saveBlockedRequests = (requests) => {
  sessionStorage.setItem(BLOCKED_REQUESTS_STORAGE_KEY, JSON.stringify(requests));
};

const getBlockedRequests = () => {
  let result;
  try {
    result = JSON.parse(sessionStorage.getItem(BLOCKED_REQUESTS_STORAGE_KEY)) || [];
  } catch (e) {
    result = [];
    sessionStorage.removeItem(BLOCKED_REQUESTS_STORAGE_KEY);
  }
  return result;
};

export const addBlockedRequest = (path) => {
  const blockedRequest = [...getBlockedRequests()];
  if (!blockedRequest.includes(path)) {
    blockedRequest.push(path);
    saveBlockedRequests(blockedRequest);
  }
};

export const removeBlockedRequest = (path) => {
  saveBlockedRequests(getBlockedRequests().filter((el) => el !== path));
};

export const isBlockedRequest = (path) => getBlockedRequests().includes(path);
