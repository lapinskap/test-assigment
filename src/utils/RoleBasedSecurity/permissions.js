export const godPermission = 'all'; // for mocks and admin admin

export const employeeAhrRolePermissionRead = 'employee:ahrRole:read';
export const employeeAhrRolePermissionWrite = 'employee:ahrRole:write';
export const employeeEmployeePermissionRead = 'employee:employee:read';
export const employeeEmployeePermissionWrite = 'employee:employee:write';
export const employeeEmployeeGroupPermissionRead = 'employee:employeeGroup:read';
export const employeeEmployeeGroupPermissionWrite = 'employee:employeeGroup:write';
export const employeeEmployeeLeavePermissionRead = 'employee:employeeLeave:read';
export const employeeEmployeeLeavePermissionWrite = 'employee:employeeLeave:write';
export const employeeOrganizationUnitPermissionRead = 'employee:organizationUnit:read';
export const employeeOrganizationUnitPermissionWrite = 'employee:organizationUnit:write';
export const employeeRentableGroupPermissionRead = 'employee:rentableGroup:read';
export const employeeRentableGroupPermissionWrite = 'employee:rentableGroup:write';
export const employeeRentableGroupSelectionWindowPermissionRead = 'employee:rentableGroupSelectionWindow:read';
export const employeeRentableGroupSelectionWindowPermissionWrite = 'employee:rentableGroupSelectionWindow:write';
export const ssoIpAddressRestrictionPermissionRead = 'sso:ipAddressRestriction:read';
export const ssoIpAddressRestrictionPermissionWrite = 'sso:ipAddressRestriction:write';
export const ssoCanSwitchUser = 'sso:canSwitchUser';
export const companyApplicationPermissionRead = 'company:application:read';
export const companyApplicationPermissionWrite = 'company:application:write';
export const companyAttachmentPermissionRead = 'company:attachment:read';
export const companyAttachmentPermissionWrite = 'company:attachment:write';
export const companyCompanyPermissionRead = 'company:company:read';
export const companyCompanyPermissionWrite = 'company:company:write';
export const companyTokenPermissionRead = 'company:token:read';
export const companyTokenPermissionWrite = 'company:token:write';
export const operatorOperatorPermissionRead = 'operator:operator:read';
export const operatorOperatorPermissionWrite = 'operator:operator:write';
export const tourismTourismObjectPermissionRead = 'tourism:tourismObject:read';
export const tourismTourismObjectPermissionWrite = 'tourism:tourismObject:write';
export const tourismTourismAttributePermissionRead = 'tourism:tourismAttribute:read';
export const tourismTourismAttributePermissionWrite = 'tourism:tourismAttribute:write';
export const tourismTourismAttributeOptionPermissionRead = 'tourism:tourismAttributeOption:read';
export const tourismTourismAttributeOptionPermissionWrite = 'tourism:tourismAttributeOption:write';
export const tourismTourismObjectChangelogPermissionRead = 'tourism:tourismObjectChangelog:read';
export const tourismTourismObjectImportAnixePermissionRead = 'tourism:tourismObjectImportAnixe:read';
export const cmsDocumentPermissionRead = 'cms:document:read';
export const cmsDocumentPermissionWrite = 'cms:document:write';
export const translationTranslateCmsPermissionWrite = 'translation:translateCms:write';
export const translationTranslateSimplePermissionWrite = 'translation:translateSimple:write';
export const translationTranslationScopePermissionRead = 'translation:translationScope:read';
export const translationTranslationScopePermissionWrite = 'translation:translationScope:write';
export const configurationConfigurationPermissionRead = 'configuration:configuration:read';
export const configurationConfigurationPermissionWrite = 'configuration:configuration:write';
export const notificationEmailPermissionWrite = 'notification:email:write';
export const notificationEmailPermissionRead = 'notification:email:read';
export const notificationPushMessagePermissionWrite = 'notification:pushMessage:write';
export const notificationPushMessagePermissionRead = 'notification:pushMessage:read';
export const notificationAppMessagePermissionWrite = 'notification:appMessage:write';
export const notificationAppMessagePermissionRead = 'notification:appMessage:read';
export const catalogProductPermissionRead = 'catalogManagement:product:read';
export const catalogProductPermissionWrite = 'catalogManagement:product:write';
export const catalogProductAttachmentPermissionRead = 'catalogManagement:attachment:read';
export const catalogProductAttachmentPermissionWrite = 'catalogManagement:attachment:write';
export const catalogCategoryPermissionRead = 'catalogManagement:businessCategory:read';
export const catalogCategoryPermissionWrite = 'catalogManagement:businessCategory:write';
export const dictionaryDictionaryPermissionRead = 'dictionary:dictionary:read';
export const dictionaryDictionaryPermissionWrite = 'dictionary:dictionary:write';
export const dictionaryDictionaryItemsImportWrite = 'dictionary:dictionaryItemsImport:write';
export const agreementAgreementPermissionRead = 'agreement:agreement:read';
export const agreementAgreementPermissionWrite = 'agreement:agreement:write';
export const subscriptionBenefitCategoryPermissionRead = 'subscriptionManagement:benefitCategory:read';
export const subscriptionBenefitCategoryPermissionWrite = 'subscriptionManagement:benefitCategory:write';
export const subscriptionBenefitPermissionRead = 'subscriptionManagement:benefit:read';
export const subscriptionBenefitPermissionWrite = 'subscriptionManagement:benefit:write';
export const subscriptionBenefitGroupPermissionRead = 'subscriptionManagement:benefitGroup:read';
export const subscriptionBenefitGroupPermissionWrite = 'subscriptionManagement:benefitGroup:write';
export const subscriptionActiveFormPermissionRead = 'subscriptionManagement:activeForm:read';
export const subscriptionActiveFormGroupPermissionWrite = 'subscriptionManagement:activeForm:write';
export const subscriptionAttachmentFilePermissionRead = 'subscriptionManagement:attachmentFile:read';
export const subscriptionAttachmentFilePermissionWrite = 'subscriptionManagement:attachmentFile:write';
export const subscriptionAttachmentPermissionRead = 'subscriptionManagement:attachment:read';
export const subscriptionAttachmentPermissionWrite = 'subscriptionManagement:attachment:write';
export const subscriptionPdfFormFilePermissionRead = 'subscriptionManagement:pdfFormFile:read';
export const subscriptionPdfFormFilePermissionWrite = 'subscriptionManagement:pdfFormFile:write';
export const subscriptionEmployeeSubscriptionItemResign = 'subscriptionManagement:employeeSubscriptionItem:resign';
export const subscriptionEmployeeSubscriptionItemCancel = 'subscriptionManagement:employeeSubscriptionItem:cancel';
export const subscriptionEmployeeSubscriptionItemAccept = 'subscriptionManagement:employeeSubscriptionItem:accept';
export const subscriptionPdfFormPermissionRead = 'subscriptionManagement:pdfForm:read';
export const subscriptionPdfFormPermissionWrite = 'subscriptionManagement:pdfForm:write';
export const subscriptionBenefitEmployeeGroupPermissionRead = 'subscriptionManagement:benefitEmployeeGroup:read';
export const subscriptionBenefitEmployeeGroupPermissionWrite = 'subscriptionManagement:benefitEmployeeGroup:write';
export const banksBanksPermissionRead = 'company:pointsBank:read';
export const banksBanksPermissionWrite = 'company:pointsBank:write';
export const employeeGroupProductConfigPermissionRead = 'employeeGroupProduct:employeeGroupProductData:read';
export const employeeGroupProductConfigPermissionWrite = 'employeeGroupProduct:employeeGroupProductData:write';
// BELOW ARE NOT CONFIRMED PERMISSIONS (WAITING FOR BACKEND
export const companyFundingPermissionRead = 'company:funding:read';
export const companyFundingPermissionWrite = 'company:funding:write';
