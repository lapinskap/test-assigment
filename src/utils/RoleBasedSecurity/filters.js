import React from 'react';
import { godPermission } from './permissions';
import __ from '../Translations';
import ForbiddenPage from './SecurityComponents/ForbiddenPage';

export const filterNavItemsByAlc = (menu, userInfo) => {
  const permissions = userInfo.getPermissions();
  return menu.map((item) => {
    if (item.aclKey && !permissions.includes(item.aclKey) && !permissions.includes(godPermission)) {
      return null;
    }
    const updatedItem = { ...item };
    if (item.content) {
      updatedItem.content = filterNavItemsByAlc(item.content, userInfo);
      if (updatedItem.content.length === 0) {
        return null;
      }
    }
    return updatedItem;
  }).filter(Boolean);
};

export const filterTabsByAcl = (tabsConfig, userInfo) => {
  const permissions = userInfo.getPermissions();
  return tabsConfig.map((item) => {
    if (item.aclKey && !permissions.includes(item.aclKey) && !permissions.includes(godPermission)) {
      return {
        ...item,
        component: <ForbiddenPage permission={item.aclKey} />,
        disabled: true,
        tabClassName: 'not-allowed',
        tabTitle: __('Nie masz uprawnieÅ„ ({0})', [item.aclKey]),
      };
    }
    return item;
  }).filter(Boolean);
};

export const hasAccessTo = (userInfo, aclKey) => {
  const permissions = userInfo.getPermissions();
  return Boolean(permissions.includes(godPermission) || permissions.includes(aclKey));
};
