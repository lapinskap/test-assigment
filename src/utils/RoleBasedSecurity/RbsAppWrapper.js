import PropTypes from 'prop-types';
import React, { useEffect, useState } from 'react';
import RbsContext from './RbsContext';
import {
  clearSessionData, deleteSession, getSession, getUserInfo, initSession,
} from './Session';
import __ from '../Translations';
import tmpAuthorization from './tmpAuthorization';
import DefaultFallback from '../../Layout/AppMain/DefaultFallback';
import UserInfo from './UserInfo';

export default function RbsAppWrapper({ children }) {
  const [userInfo, setUserInfo] = useState({});
  const [isLoggedIn, setIsLogged] = useState(false);
  const [isInitializing, setIsInitializing] = useState(true);

  useEffect(() => {
    initializeSession();
  }, []);

  const tmpLoginAction = async (type) => {
    try {
      const {
        token, refresh_token: refreshToken, date, expiresIn, userInfo: user,
      } = await tmpAuthorization(type);

      const session = getSession();
      await session.login(token, refreshToken, date, expiresIn);
      setUserInfo(new UserInfo(user));
      setIsLogged(Boolean(await session.getValidAccessToken()));
      return null;
    } catch (e) {
      console.error(e);
      return __('Nieprawidłowy login lub hasło');
    }
  };
  const logoutAction = async () => {
    try {
      clearSessionData();
      await new Promise((resolve) => {
        window.location = `/signin/oauth/logout?redirect_url=${window.location.origin}`;
        setTimeout(() => resolve(), 5000);
      });
      return null;
    } catch (e) {
      console.error(e);
      return __('Nie udaoło się wylogować');
    }
  };

  const initializeSession = async () => {
    const session = await initSession();
    let isAlreadyLoggedIn = Boolean(await session.getValidAccessToken());
    if (isAlreadyLoggedIn) {
      try {
        const user = await getUserInfo();
        setUserInfo(new UserInfo(user));
      } catch (e) {
        clearSessionData();
        deleteSession();
        await initSession();
        isAlreadyLoggedIn = false;
      }
    }

    setIsLogged(isAlreadyLoggedIn);
    setIsInitializing(false);
  };

  if (isInitializing) {
    return DefaultFallback;
  }
  return (
    <RbsContext.Provider value={{
      isLoggedIn,
      login: tmpLoginAction,
      logout: logoutAction,
      userInfo,
    }}
    >
      {children}
    </RbsContext.Provider>
  );
}

RbsAppWrapper.propTypes = {
  children: PropTypes.node.isRequired,
};
