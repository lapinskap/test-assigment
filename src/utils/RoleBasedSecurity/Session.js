// eslint-disable-next-line import/no-cycle
import { restApiRequest, setIsMockView, SSO_SERVICE } from '../Api';
import UserInfo from './UserInfo';
// eslint-disable-next-line import/no-cycle
import { MOCK_ADMIN_ACCESS_TOKEN } from './tmpAuthorization';
import { BLOCKED_REQUESTS_STORAGE_KEY } from './blockedRequests';

export const USER_INFO_STORAGE_KEY = 'user_info';
export const TMP_LOGIN_STORAGE_KEY = 'tmp_login';
export const SESSION_STORAGE_KEY = 'sso-token';

let singletonHandler = null;

class Session {
  constructor(
    accessToken = null,
    refreshToken = null,
    date = 0,
    expiresIn = 0,
  ) {
    this.setData(accessToken, refreshToken, date, expiresIn);
    this.getUserInfo = () => new UserInfo({});
  }

  updateSession(token, refreshToken, date, expiresIn) {
    const newSessionData = {
      token,
      refresh_token: refreshToken,
      date,
      expiresIn,
    };
    localStorage.setItem(SESSION_STORAGE_KEY, JSON.stringify(newSessionData));
    this.setData(token, refreshToken, date, expiresIn);
  }

  setData(accessToken, refreshToken, date, expiresIn) {
    this.getRefreshToken = () => refreshToken;
    this.getAccessTokenTtl = () => date + (expiresIn * 1000);
    this.getAccessToken = () => accessToken;
  }

  isAccessTokenValid() {
    return (Date.now() < this.getAccessTokenTtl());
  }

  async getValidAccessToken() {
    try {
      const token = this.getAccessToken();
      if (token && this.isAccessTokenValid()) {
        return token;
      }
      if (this.getRefreshToken()) {
        this.refreshToken();
        return localStorage.getItem(TMP_LOGIN_STORAGE_KEY) ? this.getAccessToken() : new Promise(() => this.refreshToken());
      }
    } catch (e) {
      console.error(e);
    }
    return '';
  }

  login(accessToken, refreshToken, date, expiresIn) {
    this.updateSession(accessToken, refreshToken, date, expiresIn);
    localStorage.setItem(TMP_LOGIN_STORAGE_KEY, '1');
  }

  // eslint-disable-next-line class-methods-use-this
  refreshToken() {
    if (!localStorage.getItem(TMP_LOGIN_STORAGE_KEY)) {
      window.location = `/signin/oauth/client/refresh?refresh_token=${
        this.getRefreshToken()
      }&referer_url=${
        encodeURIComponent(`/${window.location.hash}`)
      }`;
    } else {
      this.login(MOCK_ADMIN_ACCESS_TOKEN, MOCK_ADMIN_ACCESS_TOKEN, Date.now(), 86400);
    }
  }
}

export function clearSessionData() {
  setIsMockView(false);
  localStorage.removeItem(TMP_LOGIN_STORAGE_KEY);
  localStorage.removeItem(USER_INFO_STORAGE_KEY);
  sessionStorage.removeItem(USER_INFO_STORAGE_KEY);
  sessionStorage.removeItem(BLOCKED_REQUESTS_STORAGE_KEY);
  localStorage.removeItem(SESSION_STORAGE_KEY);
}

/**
 * @returns {Session}
 */
export function getSession() {
  if (!singletonHandler) {
    throw new Error('Session has not been initialized');
  }
  return singletonHandler;
}

// In localStorage we keep user info for tmp logins admin:admin omb:omb etc.
export async function getUserInfo() {
  let result;
  const devLoginUserInfo = localStorage.getItem(USER_INFO_STORAGE_KEY);
  if (devLoginUserInfo) {
    result = JSON.parse(devLoginUserInfo);
  } else {
    const savedUserInfoJson = sessionStorage.getItem(USER_INFO_STORAGE_KEY);
    let savedUserInfo = null;
    if (savedUserInfoJson) {
      savedUserInfo = JSON.parse(savedUserInfoJson);
    }
    const [{
      nickname: username, email, given_name: firstName, family_name: lastName, role, company_id: companyId, sub: id,
    }, { permissions, relations }] = await Promise.all([userInfoRequest(), permissionsRequest()]);
    const userData = {
      id,
      role,
      firstName,
      lastName,
      username,
      email,
      companyId,
      permissions,
      relations,
    };
    if (savedUserInfo && savedUserInfo.id !== id) {
      throw new Error('User id has changed');
    }
    sessionStorage.setItem(USER_INFO_STORAGE_KEY, JSON.stringify(userData));
    localStorage.removeItem(USER_INFO_STORAGE_KEY);
    result = userData;
  }
  return result;
}

export const userInfoRequest = async () => restApiRequest(
  SSO_SERVICE,
  '/userinfo',
  'POST',
  {
    body: {},
  },
  {
    sub: '54d539d4-d178-11ea-87d0-0242ac130003',
    name: 'jantestowy',
    email: 'jan.testowy@example.com',
    role: 'omb',
    company_id: 'e1e3fbc1-0c0a-4032-9824-f688d167acc5',
    family_name: 'Testowy',
    given_name: 'Jan',
    nickname: 'jantestowy',
    preferred_username: 'jantestowy',
  },
);
export const permissionsRequest = async () => restApiRequest(
  SSO_SERVICE,
  '/me/permissions',
  'GET',
  {},
  {
    permissions: [
      'employee:employeeGroup:read',
      'employee:employeeLeave:read',
      'sso:ipAddressRestriction:write',
      'company:attachment:write',
      'company:application:write',
      'employee:ahrRole:read',
    ],
    relations: {
      company: [],
      employeeGroup: [],
      organizationUnit: [],
      rentableGroup: [],
      administeredCompanies: [],
      administeredEmployeeGroups: [],
      administeredOrganizationUnits: [],
    },
  },
);

export function deleteSession() {
  singletonHandler = null;
}

/**
 * @returns {Session}
 */
export async function initSession() {
  let session;
  if (!singletonHandler) {
    try {
      const storedValue = localStorage.getItem(SESSION_STORAGE_KEY);
      let accessToken;
      let refreshToken;
      let date;
      let expiresIn;
      if (storedValue) {
        const parsedValue = JSON.parse(storedValue);
        accessToken = parsedValue.token;
        refreshToken = parsedValue.refresh_token;
        date = parsedValue.date;
        expiresIn = parsedValue.expires_in;
      }
      session = new Session(accessToken, refreshToken, date, expiresIn);
      singletonHandler = session;
    } catch (e) {
      console.error(e);
      session = new Session();
      singletonHandler = session;
    }
  }
  return singletonHandler;
}
