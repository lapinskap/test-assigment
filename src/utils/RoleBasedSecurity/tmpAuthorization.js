/* eslint import/no-cycle:0 */
import { setIsMockView } from '../Api';
import { USER_INFO_STORAGE_KEY } from './Session';
import { godPermission } from './permissions';

export const MOCK_ADMIN_ACCESS_TOKEN = 'asds';

const mockData = {
  token: MOCK_ADMIN_ACCESS_TOKEN,
  refresh_token: 'qwerty',
  date: Date.now(),
  expiresIn: 3600,
};

export default function tmpAuthorization(type) {
  let isMock = false;
  let companyId = 0;
  let role;
  let lastName;

  switch (type) {
    case 'omb':
      role = 'omb';
      lastName = 'mocks';
      isMock = true;
      break;
    case 'ahr':
      role = 'ahr';
      lastName = 'ahr mocks';
      isMock = true;
      companyId = 'a43275e4-eeb2-11ea-adc1-0242ac1200021';
      break;
    case 'admin':
      role = 'omb';
      lastName = 'admin';
      break;
    case 'adminahr':
      role = 'ahr';
      lastName = 'admin ahr';
      companyId = '18be83b7-79e7-4526-8752-0054419d8da9';
      break;
    default:
      return false;
  }
  setIsMockView(isMock);
  const userInfo = {
    id: 'dev',
    role,
    companyId,
    firstName: 'dev',
    lastName,
    email: 'test@example.com',
    permissions: [godPermission],
    relations: {
      company: companyId,
    },
    username: type,
  };

  window.localStorage.setItem(USER_INFO_STORAGE_KEY, JSON.stringify(userInfo));
  return {
    ...mockData,
    userInfo,
  };
}
