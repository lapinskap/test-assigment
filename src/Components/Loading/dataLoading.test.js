import React, { useState } from 'react';
import '../../tests/setupTests';
import EnzymeToJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import DataLoading from './dataLoading';
import { waitToLoadMocks } from '../../tests/helpers';
import { initSession, deleteSession } from '../../utils/RoleBasedSecurity/Session';

describe('<DataLoading />', () => {
  const mockEndpoint = '/company/companyData/edit';
  const service = '';

  afterEach(() => {
    window.SKIP_TEST_MODE = false;
    deleteSession();
    jest.clearAllMocks();
  });

  it('renders correctly with mocks', async () => {
    global.fetch = jest.fn(() => Promise.resolve({
      ok: true,
      json: () => Promise.resolve({}),
    }));
    const Component = () => {
      const [fetched, setFetched] = useState(false);
      return (
        <DataLoading
          service={service}
          updateData={() => {
            setFetched(true);
          }}
          endpoint={mockEndpoint}
          isMock
          fetchedData={fetched}
          forceLoader={false}
          isNew={false}
          mockDataEndpoint={mockEndpoint}
        >
          <div>Content</div>
        </DataLoading>
      );
    };

    const wrapper = mount(<Component />);
    await waitToLoadMocks();
    wrapper.update();
    expect(EnzymeToJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly new', async () => {
    const wrapper = mount(
      <DataLoading
        service={service}
        updateData={() => {}}
        endpoint={mockEndpoint}
        isMock
        fetchedData={false}
        isNew
        forceLoader={false}
        mockDataEndpoint={mockEndpoint}
      >
        <div>Content</div>
      </DataLoading>,
    );
    wrapper.update();
    expect(EnzymeToJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly whith 500 error fetch', async () => {
    window.SKIP_TEST_MODE = true;
    global.fetch = jest.fn(() => Promise.resolve({
      ok: false,
      status: 500,
      json: () => Promise.resolve({}),
    }));
    initSession();
    const updateDataSpy = jest.fn();
    const wrapper = mount(
      <DataLoading
        service={service}
        updateData={updateDataSpy}
        endpoint={mockEndpoint}
        fetchedData={false}
        forceLoader={false}
        isNew={false}
      >
        <div>Content</div>
      </DataLoading>,
    );
    wrapper.update();
    expect(EnzymeToJson(wrapper)).toMatchSnapshot();
    expect(updateDataSpy).toHaveBeenCalledTimes(0);
  });

  it('renders correctly whith 403 error fetch', async () => {
    window.SKIP_TEST_MODE = true;
    global.fetch = jest.fn(() => Promise.resolve({
      ok: false,
      status: 403,
      json: () => Promise.resolve({}),
    }));
    initSession();
    const updateDataSpy = jest.fn();
    const wrapper = mount(
      <DataLoading
        service={service}
        updateData={updateDataSpy}
        endpoint={mockEndpoint}
        fetchedData={false}
        forceLoader={false}
        isNew={false}
      >
        <div>Content</div>
      </DataLoading>,
    );
    wrapper.update();
    expect(EnzymeToJson(wrapper)).toMatchSnapshot();
    expect(updateDataSpy).toHaveBeenCalledTimes(0);
  });
});
