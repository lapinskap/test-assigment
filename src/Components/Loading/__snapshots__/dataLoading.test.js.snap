// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataLoading /> renders correctly new 1`] = `
<DataLoading
  endpoint="/company/companyData/edit"
  fetchedData={false}
  forceLoader={false}
  isMock={true}
  isNew={true}
  mockDataEndpoint="/company/companyData/edit"
  options={Object {}}
  service=""
  updateData={[Function]}
>
  <ContentLoading
    backgroundStyle={null}
    message={
      <Loader
        active={true}
        color="#545cd8"
        type="line-scale"
      />
    }
    messageStyle={
      Object {
        "position": "absolute",
        "right": "50%",
        "top": "20%",
      }
    }
    show={false}
    useBlur={true}
  >
    <Loader
      backgroundStyle={null}
      contentBlur={4}
      message={
        <Loader
          active={true}
          color="#545cd8"
          type="line-scale"
        />
      }
      messageStyle={
        Object {
          "position": "absolute",
          "right": "50%",
          "top": "20%",
        }
      }
      priority={0}
      show={false}
    >
      <div
        className="Loader"
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <div
          className="Loader__content"
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <div>
            Content
          </div>
        </div>
      </div>
    </Loader>
  </ContentLoading>
</DataLoading>
`;

exports[`<DataLoading /> renders correctly whith 403 error fetch 1`] = `
<DataLoading
  endpoint="/company/companyData/edit"
  fetchedData={false}
  forceLoader={false}
  isMock={false}
  isNew={false}
  mockDataEndpoint={null}
  options={Object {}}
  service=""
  updateData={[MockFunction]}
>
  <ContentLoading
    backgroundStyle={null}
    message={
      <Loader
        active={true}
        color="#545cd8"
        type="line-scale"
      />
    }
    messageStyle={
      Object {
        "position": "absolute",
        "right": "50%",
        "top": "20%",
      }
    }
    show={true}
    useBlur={true}
  >
    <Loader
      backgroundStyle={null}
      contentBlur={4}
      message={
        <Loader
          active={true}
          color="#545cd8"
          type="line-scale"
        />
      }
      messageStyle={
        Object {
          "position": "absolute",
          "right": "50%",
          "top": "20%",
        }
      }
      priority={0}
      show={true}
    >
      <div
        className="Loader"
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <div
          className="Loader__content"
          style={
            Object {
              "MozFilter": "blur(4px)",
              "OFilter": "blur(4px)",
              "WebkitFilter": "blur(4px)",
              "filter": "blur(4px)",
              "msFilter": "blur(4px)",
              "opacity": 1,
            }
          }
        >
          <div>
            Content
          </div>
        </div>
        <div
          className="Loader__background"
          style={
            Object {
              "backgroundColor": "rgba(0,0,0,0.5)",
              "display": "block",
              "height": "100%",
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": "100%",
              "zIndex": 10,
            }
          }
        >
          <div
            className="Loader__foreground"
            style={
              Object {
                "color": "white",
                "display": "table",
                "height": "100%",
                "textAlign": "center",
                "width": "100%",
                "zIndex": 20,
              }
            }
          >
            <div
              className="Loader__message"
              style={
                Object {
                  "display": "table-cell",
                  "position": "absolute",
                  "right": "50%",
                  "top": "20%",
                  "verticalAlign": "middle",
                }
              }
            >
              <Loader
                active={true}
                color="#545cd8"
                type="line-scale"
              >
                <div
                  className="loader loader-undefined loader-active"
                >
                  <div
                    className="loader-inner line-scale"
                  >
                    <div
                      key="0"
                      style={
                        Object {
                          "backgroundColor": "#545cd8",
                        }
                      }
                    />
                    <div
                      key="1"
                      style={
                        Object {
                          "backgroundColor": "#545cd8",
                        }
                      }
                    />
                    <div
                      key="2"
                      style={
                        Object {
                          "backgroundColor": "#545cd8",
                        }
                      }
                    />
                    <div
                      key="3"
                      style={
                        Object {
                          "backgroundColor": "#545cd8",
                        }
                      }
                    />
                    <div
                      key="4"
                      style={
                        Object {
                          "backgroundColor": "#545cd8",
                        }
                      }
                    />
                  </div>
                </div>
              </Loader>
            </div>
          </div>
        </div>
      </div>
    </Loader>
  </ContentLoading>
</DataLoading>
`;

exports[`<DataLoading /> renders correctly whith 500 error fetch 1`] = `
<DataLoading
  endpoint="/company/companyData/edit"
  fetchedData={false}
  forceLoader={false}
  isMock={false}
  isNew={false}
  mockDataEndpoint={null}
  options={Object {}}
  service=""
  updateData={[MockFunction]}
>
  <ContentLoading
    backgroundStyle={null}
    message={
      <Loader
        active={true}
        color="#545cd8"
        type="line-scale"
      />
    }
    messageStyle={
      Object {
        "position": "absolute",
        "right": "50%",
        "top": "20%",
      }
    }
    show={true}
    useBlur={true}
  >
    <Loader
      backgroundStyle={null}
      contentBlur={4}
      message={
        <Loader
          active={true}
          color="#545cd8"
          type="line-scale"
        />
      }
      messageStyle={
        Object {
          "position": "absolute",
          "right": "50%",
          "top": "20%",
        }
      }
      priority={0}
      show={true}
    >
      <div
        className="Loader"
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <div
          className="Loader__content"
          style={
            Object {
              "MozFilter": "blur(4px)",
              "OFilter": "blur(4px)",
              "WebkitFilter": "blur(4px)",
              "filter": "blur(4px)",
              "msFilter": "blur(4px)",
              "opacity": 1,
            }
          }
        >
          <div>
            Content
          </div>
        </div>
        <div
          className="Loader__background"
          style={
            Object {
              "backgroundColor": "rgba(0,0,0,0.5)",
              "display": "block",
              "height": "100%",
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": "100%",
              "zIndex": 10,
            }
          }
        >
          <div
            className="Loader__foreground"
            style={
              Object {
                "color": "white",
                "display": "table",
                "height": "100%",
                "textAlign": "center",
                "width": "100%",
                "zIndex": 20,
              }
            }
          >
            <div
              className="Loader__message"
              style={
                Object {
                  "display": "table-cell",
                  "position": "absolute",
                  "right": "50%",
                  "top": "20%",
                  "verticalAlign": "middle",
                }
              }
            >
              <Loader
                active={true}
                color="#545cd8"
                type="line-scale"
              >
                <div
                  className="loader loader-undefined loader-active"
                >
                  <div
                    className="loader-inner line-scale"
                  >
                    <div
                      key="0"
                      style={
                        Object {
                          "backgroundColor": "#545cd8",
                        }
                      }
                    />
                    <div
                      key="1"
                      style={
                        Object {
                          "backgroundColor": "#545cd8",
                        }
                      }
                    />
                    <div
                      key="2"
                      style={
                        Object {
                          "backgroundColor": "#545cd8",
                        }
                      }
                    />
                    <div
                      key="3"
                      style={
                        Object {
                          "backgroundColor": "#545cd8",
                        }
                      }
                    />
                    <div
                      key="4"
                      style={
                        Object {
                          "backgroundColor": "#545cd8",
                        }
                      }
                    />
                  </div>
                </div>
              </Loader>
            </div>
          </div>
        </div>
      </div>
    </Loader>
  </ContentLoading>
</DataLoading>
`;

exports[`<DataLoading /> renders correctly with mocks 1`] = `
<Component>
  <DataLoading
    endpoint="/company/companyData/edit"
    fetchedData={true}
    forceLoader={false}
    isMock={true}
    isNew={false}
    mockDataEndpoint="/company/companyData/edit"
    options={Object {}}
    service=""
    updateData={[Function]}
  >
    <ContentLoading
      backgroundStyle={null}
      message={
        <Loader
          active={true}
          color="#545cd8"
          type="line-scale"
        />
      }
      messageStyle={
        Object {
          "position": "absolute",
          "right": "50%",
          "top": "20%",
        }
      }
      show={false}
      useBlur={true}
    >
      <Loader
        backgroundStyle={null}
        contentBlur={4}
        message={
          <Loader
            active={true}
            color="#545cd8"
            type="line-scale"
          />
        }
        messageStyle={
          Object {
            "position": "absolute",
            "right": "50%",
            "top": "20%",
          }
        }
        priority={0}
        show={false}
      >
        <div
          className="Loader"
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            className="Loader__content"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <div>
              Content
            </div>
          </div>
        </div>
      </Loader>
    </ContentLoading>
  </DataLoading>
</Component>
`;
